socorro/external/crashstorage_base.py

```diff
@@ -253,368 +242,65 @@ def remove(self, crash_id):
-        self.stores = DotDict()
-        for storage_namespace in self.storage_namespaces:
-            absolute_namespace = ".".join(
-                x for x in [namespace, storage_namespace] if x
-            )
-            self.stores[storage_namespace] = config[
-                storage_namespace
-            ].crashstorage_class(
-                config[storage_namespace],
-                namespace=absolute_namespace,
-            )
-
-    def close(self):
-        """Close resources used by crashstorage instances.
-
-        :raises PolyStorageError: An exception container holding a list of the
-            exceptions raised by the subordinate storage
-            systems.
-
-        """
-        storage_exception = PolyStorageError()
-        for a_store in self.stores.values():
-            try:
-                a_store.close()
-            except Exception as x:
-                self.logger.error("%s failure: %s", a_store.__class__, str(x))
-                storage_exception.gather_current_exception()
-        if storage_exception.has_exceptions():
-            raise storage_exception
-
-    def save_raw_crash(self, raw_crash, dumps, crash_id):
-        """Save raw crash to all crashstorage destinations.
-
-        :param raw_crash: the raw crash data
-        :param dumps: mapping of dump name keys -> dump binary
-        :param crash_id: the crash report id
-
-        """
-        storage_exception = PolyStorageError()
-        for a_store in self.stores.values():
-            try:
-                a_store.save_raw_crash(raw_crash, dumps, crash_id)
-            except Exception as x:
-                self.logger.error("%s failure: %s", a_store.__class__, str(x))
-                storage_exception.gather_current_exception()
-        if storage_exception.has_exceptions():
-            raise storage_exception
-
-    def save_processed_crash(self, raw_crash, processed_crash):
-        """Save processed crash to all crashstorage destinations
-
-        :param raw_crash: the raw crash data
-        :param processed_crash: the processed crash data
-
-        NOTE(willkg): Crash storage systems that implement this should
-        not mutate the raw and processed crash structures!
-
-        """
-        storage_exception = PolyStorageError()
-        for a_store in self.stores.values():
-            try:
-                a_store.save_processed_crash(raw_crash, processed_crash)
-            except Exception:
-                store_class = getattr(a_store, "wrapped_object", a_store.__class__)
-                crash_id = processed_crash.get("uuid", "NONE")
-                self.logger.error(
-                    "%r failed (crash id: %s)", store_class, crash_id, exc_info=True
-                )
-                storage_exception.gather_current_exception()
-        if storage_exception.has_exceptions():
-            raise storage_exception
-
-
-class BenchmarkingCrashStorage(CrashStorageBase):
```

Comment:
I removed this. The only thing we used it for was gathering timing metrics for `save_processed_crash` in the crashstorage destinations. It's easier to assume we want that and add a couple of lines to the processor code that's saving the crash data in all the destinations.